File: peer-prep-be/src/models/question_model.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Question struct {
	QuestionId primitive.ObjectID `json:"questionId,omitempty"`
	QuestionTitle string `json:"questionTitle" validate:"required"`
	QuestionDescription string `json:"questionDescription" validate:"required"`
	QuestionCategory string `json:"questionCategory" validate:"required"`
	QuestionComplexity string `json:"questionComplexity" validate:"required"`
}

File: peer-prep-fe/src/app/models/question.model.ts
// export interface Question {
//   question_id: string
//   question_title: string
//   question_description: string
//   question_categories: string[]
//   question_complexity: string
// }

export interface Example {
  input: string;
  output: string;
}

export interface Question {
  question_id: string;
  question_title: string;
  question_description: string;
  question_categories: string[];
  question_complexity: string;
  examples: Example[]; // Add examples
  constraints: string[]; // Add constraints
}
